### ISSUE TITLE
OnApplyTemplate is called no matter its initial visibility

### ISSUE THREAD

**Issue by takla21** [unoplatform/uno#13019]
### Current behavior

`OnApplyTemplate` is called no matter its visibility.

Example from the sample below
```
Navigating to blankpage. IsVisible = True
TestControl 0 created!
Apply template TestControl 0
Loaded TestControl 0
Unloaded TestControl 0
Navigating to blankpage. IsVisible = False
TestControl 1 created!
Apply template TestControl 1
Loaded TestControl 1
```

For the given custom control
```cs
public partial class TestControl : Control
{
   private static int _cpt = 0;

    private readonly string _id;

    public TestControl()
    {
        _id = "TestControl " + _cpt++;

        System.Diagnostics.Debug.WriteLine(_id + " created!");

        Loaded += TestControl_Loaded;
        Unloaded += TestControl_Unloaded;
    }

    private void TestControl_Unloaded(object sender, RoutedEventArgs e)
    {
        System.Diagnostics.Debug.WriteLine("Unloaded " + _id);
    }

    private void TestControl_Loaded(object sender, RoutedEventArgs e)
    {
        System.Diagnostics.Debug.WriteLine("Loaded " + _id);
    }

    protected override void OnApplyTemplate()
    {
        base.OnApplyTemplate();

        System.Diagnostics.Debug.WriteLine("Apply template " + _id);
    }
}
```

### Expected behavior

On winui, when you navigate to a page with a control collapsed, `OnApplyTemplate` is not called.

Example from the sample below
```
Navigating to blankpage. IsVisible = True
TestControl 0 created!
Apply template TestControl 0
Loaded TestControl 0
Unloaded TestControl 0
Navigating to blankpage. IsVisible = False
TestControl 1 created!
Loaded TestControl 1
```

### How to reproduce it (as minimally and precisely as possible)

- Download this sample [OnLoadedSample.zip](https://github.com/unoplatform/uno/files/12175026/OnLoadedSample.zip)
- Deploy on android/iOS
- UnCheck checkbox
- Press the navigate button
- Check logs

![ezgif com-video-to-gif](https://github.com/unoplatform/uno/assets/17461593/9b56a356-e939-4367-a0c6-a98a7cd38951)

### Workaround

N/A

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

5.0.0-dev.1614

### Affected platforms

Android, iOS

### IDE

Visual Studio 2022

### IDE version

17.6.1

### Anything else we need to know?

Maybe it's related to #2895?

**Comment by Youssef1313** [comment]
https://github.com/unoplatform/uno/issues/3519

>  On UWP, OnApplyTemplate() is called during the first measure phase (step 6). Currently on Uno (all platforms) it's called synchronously with the assignation to the Template (ControlTemplate) property (step 4)

**Comment by Youssef1313** [comment]
There is also this: https://github.com/unoplatform/uno/issues/190

**Comment by takla21** [comment]
@Youssef1313 I tried with the flag you mentioned and it's still the same result

```cs
public AppHead()
        {
#if __ANDROID__ || __IOS__
			Uno.UI.FeatureConfiguration.Control.UseDeferredOnApplyTemplate = true;
#endif

            this.InitializeComponent();
        }
```

```
[0:] Navigating to blankpage. IsVisible = True
[0:] TestControl 0 created!
[0:] Apply template TestControl 0
[0:] Loaded TestControl 0
[0:] Unloaded TestControl 0
[0:] Navigating to blankpage. IsVisible = False
[0:] TestControl 1 created!
[0:] Apply template TestControl 1
[0:] Loaded TestControl 1
```

**Comment by jeromelaban** [comment]
Duplicate of #190

**Comment by jeromelaban** [comment]
This is an issue we still having to fix, let's track with #190
